modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java

```diff
@@ -459,79 +449,59 @@ private void requestPartitions(final RebalanceFuture fut, GridDhtPreloaderAssign
 
                 U.log(log, "Prepared rebalancing [grp=" + grp.cacheOrGroupName()
                     + ", mode=" + cfg.getRebalanceMode() + ", supplier=" + node.id() + ", partitionsCount=" + parts.size()
-                    + ", topVer=" + fut.topologyVersion() + ", localParallelism=" + locStripes
-                    + ", rmtParallelism=" + rmtStripes + ", parallelism=" + rmtTotalStripes + "]");
+                    + ", topVer=" + fut.topologyVersion() + "]");
             }
 
-            final List<IgniteDhtDemandedPartitionsMap> stripePartitions = new ArrayList<>(stripes);
-            for (int i = 0; i < stripes; i++)
-                stripePartitions.add(new IgniteDhtDemandedPartitionsMap());
+            if (!parts.isEmpty()) {
```

Comment:
I'd recommend here to use such way:

```
if (parts.isEmpty())
   return;
```

Reply:
This will cause code refactoring and merge issues.
Since goal it to perform merge to older versions, that's not a good idea to refactor anything.

Reply:
Why we should keep such a thing in the master branch only for `not having problems with cherry-pick`? I think we can do it here.